sklearn/linear_model/ridge.py

```diff
@@ -882,9 +897,14 @@ def _errors_and_values_svd_helper(self, alpha, y, v, U, UT_y):
         """Helper function to avoid code duplication between self._errors_svd
         and self._values_svd.
         """
-        w = ((v + alpha) ** -1) - (alpha ** -1)
-        c = np.dot(U, self._diag_dot(w, UT_y)) + (alpha ** -1) * y
-        G_diag = self._decomp_diag(w, U) + (alpha ** -1)
+        dummy_column = np.var(U, 0) < 1.e-12
+        # detect columns colinear to ones
+        w = alpha * ((v + alpha) ** -1) - 1
+        w[dummy_column] = - 1  # cancel the regularization for the intercept
+        c = np.dot(U, self._diag_dot(w, UT_y)) + y
+        G_diag = self._decomp_diag(w, U) + 1
+        c /= alpha  # for compatibility with textbook version
+        G_diag /= alpha  # for compatibility with textbook version
```

Comment:
At some point, I was fighting with numerical issues when one of the singular values v is 0 and alpha very small. I am not sure whether this is actually the case. I am sure, however, that the result is identical to the previous version.